cache/lru/table.go

```diff
@@ -0,0 +1,67 @@
+package lru
+
+import (
+	"reflect"
+	"sync"
+
+	"github.com/golang/groupcache/lru"
+	"github.com/renproject/kv/db"
+)
+
+type lruTable struct {
+	mu    *sync.Mutex
+	lru   *lru.Cache
+	table db.Table
+}
+
+func NewLruTable(table db.Table, maxEntries int) db.Table {
+	return &lruTable{
+		mu:    new(sync.Mutex),
+		lru:   lru.New(maxEntries),
+		table: table,
+	}
+}
+
+func (table *lruTable) Insert(key string, value interface{}) error {
+	table.mu.Lock()
+	table.lru.Add(key, value)
+	table.mu.Unlock()
+
+	return table.table.Insert(key, value)
+}
+
+func (table *lruTable) Get(key string, value interface{}) error {
+	table.mu.Lock()
+	val, ok := table.lru.Get(key)
+	table.mu.Unlock()
```

Comment:
You can use continue to use `defer` and keep it scoped by using a closure pattern.

```go
val, ok := func() (interface{}, error) {
    table.mu.Lock()
    defer table.mu.Unlock()
    return table.lru.Get(key)
}()
```